### ISSUE TITLE
Disabling collections that user doesn't have correct privilege on

### ISSUE THREAD

**Issue by tuxpiper** [ushahidi/platform#4739]
If a user doesn't have the correct privilege they should be able to perform certain action

- [X] User should not be able to view the collection when read privilege is unavailable
- [X] User should not be able to add a post to that collection when `update` privilege is unavailable

**Comment by Angamanga** [comment]
@mannyAndem Yes, please go ahead!
